addon/-private/system/class-meta.js

```diff
@@ -0,0 +1,57 @@
+import EmptyObject from 'ember-data/-private/system/empty-object';
+
+export default class ClassMeta {
+  constructor(store, modelName) {
+    this.modelName = modelName;
+    this._store = store;
+    this._modelClass = null;
+    this._idToRecord = null;
+    this._records = null;
+    this._metadata = null;
+  }
+
+  get idToRecord() {
+    return this._idToRecord || (this._idToRecord = new EmptyObject());
```

Comment:
I'm on board with the new direction, I intentionally shaped this class to follow the pojo we were currently using as the typeMap in order to limit the scope of the changes. Can we agree that this is an immediate follow up PR? Or should I go ahead and do this refactor in this PR.